csrc/compute_at_map.cpp

```diff
@@ -788,25 +788,24 @@ void ComputeAtMap::allocateIndexVariables() {
   //  all lowered kir::ForLoop will correspond to one of the disjoint sets
   //  and we only need one index variable for each set.
   for (const auto& loop_disjoint_set : id_graph_.loopNodes().disjointSets()) {
-    ParallelType ptype;
+    ParallelType ptype = ParallelType::Serial;
     // first allocate thread and grid parallel indices:
     //  The validation pass will check that the parallel bindings within the
     //  loop nodes are consistent so all the loops within this disjoint set
     //  will be realized implicitly using parallel index variables.
-    if (std::any_of(
-            loop_disjoint_set->vector().begin(),
-            loop_disjoint_set->vector().end(),
-            [&ptype](IterDomain* id) {
-              if (id->isThread() &&
-                  // Halo extended parallel loops currently are handled
-                  // differently and an index variable would still
-                  // be allocated in this case.
-                  (GpuLower::current()->haloInfo()->getExtent(id) == nullptr)) {
-                ptype = id->getParallelType();
-                return true;
-              }
-              return false;
-            })) {
+
```

Comment:
Major Change:
`Error (CLANGTIDY) [bugprone-assignment-in-if-condition,-warnings-as-errors] an assignment within an 'if' condition is bug-prone`

Reply:
Does the assignment refer to this at line 805?

```
ptype = id->getParallelType();
```